services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/long_running_tasks.py

```diff
@@ -0,0 +1,312 @@
+CONCURRENCY_STATE_SAVE_RESTORE: Final[int] = 2
+
+
+async def task_create_service_containers(
+    progress: TaskProgress,
+    settings: ApplicationSettings,
+    containers_create: ContainersCreate,
+    shared_store: SharedStore,
+    mounted_volumes: MountedVolumes,
+    app: FastAPI,
+    application_health: ApplicationHealth,
+    rabbitmq: RabbitMQ,
+    long_running_compose_timeout: int,
+    validation_timeout: int,
+) -> list[str]:
+    progress.publish(message="validating service spec", percent=0)
+
+    shared_store.compose_spec = await validate_compose_spec(
+        settings=settings,
+        compose_file_content=containers_create.docker_compose_yaml,
+        mounted_volumes=mounted_volumes,
+        docker_compose_config_timeout=validation_timeout,
+    )
+    shared_store.container_names = assemble_container_names(shared_store.compose_spec)
+
+    logger.debug("Validated compose-spec:\n%s", f"{shared_store.compose_spec}")
+
+    await send_message(rabbitmq, "starting service containers")
+    assert shared_store.compose_spec  # nosec
+
+    with directory_watcher_disabled(app):
+        # removes previous pending containers
+        progress.publish(message="cleanup previous used resources")
+        await docker_compose_rm(shared_store.compose_spec, settings)
+
+        progress.publish(message="pulling images", percent=0.01)
+        await docker_compose_pull(
+            shared_store.compose_spec, settings, timeout=long_running_compose_timeout
+        )
+
+        progress.publish(message="starting service containers", percent=0.90)
+        r = await docker_compose_up(
+            shared_store.compose_spec, settings, timeout=long_running_compose_timeout
+        )
+
+    message = f"Finished docker-compose up with output\n{r.message}"
+
+    if r.success:
+        await send_message(rabbitmq, "service containers started")
+        logger.info(message)
+        for container_name in shared_store.container_names:
+            await start_log_fetching(app, container_name)
+    else:
+        application_health.is_healthy = False
+        application_health.error_message = message
+        logger.error("Marked sidecar as unhealthy, see below for details\n:%s", message)
+        await send_message(rabbitmq, "could not start service containers")
+
+    progress.publish(message="done", percent=1)
+
+    return shared_store.container_names
+
+
+async def task_runs_docker_compose_down(
+    progress: TaskProgress,
+    app: FastAPI,
+    shared_store: SharedStore,
+    settings: ApplicationSettings,
+    command_timeout: int,
+) -> None:
+    if shared_store.compose_spec is None:
+        raise RuntimeError("No compose-spec was found")
+
+    progress.publish(message="running docker-compose-down", percent=0)
+    result = await docker_compose_down(
+        shared_store.compose_spec,
+        settings,
+        timeout=min(command_timeout, settings.DYNAMIC_SIDECAR_STOP_AND_REMOVE_TIMEOUT),
```

Comment:
timeout. what happens if the timeout triggers? does it stop removing the containers?

Reply:
I've refactored this. We use the default internal timeout.